test/api/v3/integration/groups/POST-groups_groupId_join.js

```diff
@@ -0,0 +1,63 @@
+import {
+  generateUser,
+} from '../../../../helpers/api-integration.helper';
+
+describe('POST /group/:groupId/join', () => {
+  let user;
+
+  beforeEach(() => {
+    return generateUser().then((generatedUser) => {
+      user = generatedUser;
+    });
+  });
+
+  context('Guilds', () => {
+    it('inviting to guild doesn\'t give basilist to inviter', () => {
+      let groupName = 'Test Guild';
+      let groupType = 'guild';
+
+      return generateUser({balance: 1, 'items.quests.basilist': 0 }).then((generatedUser) => {
+        user = generatedUser;
+        return user.post('/groups', {
+          name: groupName,
+          type: groupType
+        });
+      })
+      .then((guild) => {
+        return generateUser({ 'invitations.guilds': [{ id: guild._id}]}).then((generatedUser) => {
+          return generatedUser.post('/groups/' + guild._id + '/join');

```

Comment:
Yea, I agree. And, @crookedneighbor, I were discussing it. I couldn't think of the word, but Mocking was what I was thinking of. There is a difference between integration tests and unit tests though. Unit tests you definitely want to mock all this.